src/main/java/svydovets/util/BeanNameResolver.java

```diff
@@ -2,12 +2,42 @@
 
 import java.lang.reflect.Method;
 
+import svydovets.core.annotation.Bean;
+import svydovets.core.annotation.Component;
+import svydovets.core.annotation.Configuration;
+
 public class BeanNameResolver {
-    public static String resolveBeanNameByBeanType(Class<?> beanClass) {
-        return beanClass.getSimpleName();
+
+    //create javadoc for description
+    public static String resolveBeanName(Class<?> beanClass) {
+        if (beanClass.isAnnotationPresent(Configuration.class)) {
+            var beanValue = beanClass.getAnnotation(Configuration.class).value();
+            if (!beanValue.isEmpty()) {
+                return beanValue;
+            }
+        }
+        if (beanClass.isAnnotationPresent(Component.class)) {
+            var beanValue = beanClass.getAnnotation(Component.class).value();
+            if (!beanValue.isEmpty()) {
+                return beanValue;
+            }
+        }
+        return setFirstCharacterToLowercase(beanClass.getSimpleName());
```

Comment:
We can rename it to `toLoweCaseFirstCharacter`. I think it's okay to have the name with `to`. You can check the String class -> https://docs.oracle.com/javase/8/docs/api/java/lang/String.html